gcm/client.go

```diff
@@ -0,0 +1,88 @@
+package gcm
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+// gcm Client const variables
+const (
+	GCMClientTimeout = time.Second * 10
+	GCMEndpoint      = "https://fcm.googleapis.com/fcm/send"
```

Comment:
Timeout and Endpoint should be configurable. It is useful for tuning and testing.